### DISCUSSION TITLE
can add read_sql like sparksql?

### DISCUSSION THREAD

**Discussion by wonb168** [duckdb/duckdb#4548] [General]
In my opinion , duckdb is like a simple spark,
spark can write sparksql ,duckdb also.
and spark can also use read_jdbc and return a dataframe,
but duckdb no read_jdbc, if use pandas.read_sql, some column types changed, and hard to extract,
for example postgre's json and array column
[1,3,4]
[{'a':1,'b':'x',},{'a':2,'b':'y'}]

**Comment by hannes** [comment]
thanks, we could add a read_jdbc extension indeed but its not on the immediate roadmap.

**Reply by wonb168** [reply]
Most data are in rdb, if duckdb can easy read and write rdb , duckdb can act as a mini Sparksql

**Reply by Mause** [reply]
DuckDB can already read from postgres directly: https://github.com/duckdblabs/postgresscanner

**Reply by wonb168** [reply]
postgresscanner not good. It need compile with the certain version postgre, but spark's read_jdbc can read many versions depond on the jar, and also it no need to compile and install.
So, it's much better if duckdb has read_sql or read_jdbc funtion itself.

**Reply by Alex-Monahan** [reply]
The postgresscanner does not need to be recompiled for a specific version of Postgres.

**Reply by Mause** [reply]
> postgresscanner not good. It need compile with the certain version postgre, but spark's read_jdbc can read many versions depond on the jar, and also it no need to compile and install.
> So, it's much better if duckdb has read_sql or read_jdbc funtion itself.

We could probably add a read_jdbc function or similar, as part of the JDBC client, similar to how we've integrated with pandas in the Python client. If you wanted this feature you could submit a pr?
